cmd/kubectl-kubestellar/cmd/ensure/location.go

```diff
@@ -0,0 +1,166 @@
+)
+
+var imw string // IMW name, provided by --imw flag
+
+// Create the Cobra sub-command for 'kubectl kubestellar ensure location'
+func newCmdEnsureLocation(cliOpts *genericclioptions.ConfigFlags) *cobra.Command {
+	// Make location command
+	cmdLocation := &cobra.Command{
+		Use:     "location --imw <IMW_NAME> <LOCATION_NAME> <LABEL_1 ...>",
+		Aliases: []string{"loc"},
+		Short:   "Ensure existence and configuration of an inventory listing for a WEC",
+		// We actually require at least 2 arguments (location name and a label),
+		// but more descriptive error messages will be provided by leaving this
+		// set to 1.
+		Args: cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// At this point set silence usage to true, so that any errors
+			// following do not result in the help being printed. We only
+			// want the help to be displayed when the error is due to an
+			// invalid command.
+			cmd.SilenceUsage = true
+			err := ensureLocation(cmd, args, cliOpts)
+			return err
+		},
+	}
+
+	// Add flag for IMW name
+	cmdLocation.Flags().StringVar(&imw, "imw", "", "IMW name")
+	cmdLocation.MarkFlagRequired("imw")
+	return cmdLocation
+}
+
+// The IMW name is provided by the --imw flag (stored in the "imw" string
+// variable), and the location name is a command line argument.
+// Labels to check are provided as additional arguments in key=value pairs.
+// In this function we will:
+// - Work in the provided IMW
+// - Check if APIBinding "edge.kubestellar.io" exists in IMW, create if not
+// - Check for SyncTarget of provided name in IMW, create if not
+// - Check that SyncTarget has an "id" label matching the Location name
+// - Ensure that SyncTarget has the labels provided by the user
+// - Check for Location of provided name in IMW, create if not
+// - Ensure that Location has the labels provided by the user
+// - If Location "default" exists, delete it
+func ensureLocation(cmdLocation *cobra.Command, args []string, cliOpts *genericclioptions.ConfigFlags) error {
+	locationName := args[0]
+	labels := args[1:]
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+
+	// Print all flags and their values if verbosity level is at least 1
+	cmdLocation.Flags().VisitAll(func(flg *pflag.Flag) {
+		logger.V(1).Info(fmt.Sprintf("Command line flag %s=%s", flg.Name, flg.Value))
+	})
+
+	// Make sure user provided location name is valid
+	err := plugin.CheckLocationName(locationName)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Problem with location name %s", locationName))
+		return err
+	}
+
+	// Make sure user provided labels are valid
+	err = plugin.CheckLabelArgs(labels)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Problem with label arguments %s", labels))
+		return err
+	}
+
+	// Set context to root, later on we will append the IMW name to the root server
+	configContext := "root"
+	cliOpts.Context = &configContext
```

Comment:
This unconditionally sets the context. The normal behavior is to let the user control the context through the command line flag. To do that, set `cliOpts.Context` _before_ calling `cliOpts.AddFlags(fs)`.